### PR TITLE
next/667/20241213/v1

### PR DESCRIPTION
**Pull request by victorjulien**
Staging:
 - #12275

### CODE REVIEW

File: rust/src/dns/parser.rs [hunk 1]
```diff
@@ -1120,21 +1119,7 @@ mod tests {
...
-        // Followed by a pointer back to the beginning.
-        buf.push(0b1100_0000);
-        buf.push(0b0000_0000);
-
-        // The start of the name, which is pointer to the beginning of
-        // the buffer.
-        buf.push(0b1100_0000);
-        buf.push(0b000_0000);
-
+        let buf: Vec<u8> = vec![0b0000_0001, b'a', 0b1100_0000, 0b0000_0000, 0b1100_0000, 0b000_0000];
```
Comment (jasonish) [inline rust/src/dns/parser.rs:1122]: This loses the documentation of the test. It would be better to suppress the warning here I think.

**Comment by jasonish** [comment]
Version with comments retained: https://github.com/OISF/suricata/pull/12284

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23937

**Comment by victorjulien** [comment]
Redoing to get an improved clippy fix in.

**Pull request closed without merging**